contracts/proxy-lib/tests/common/proxy_lib_helper.rs

```diff
@@ -22,13 +23,13 @@ impl ProxyContractHelper {
 
     pub fn create_proposal_request(
         &self,
-        id: &ProposalId,
+        id: &Repr<ProposalId>,
         author: &SigningKey,
         actions: &Vec<ProposalAction>,
     ) -> eyre::Result<Signed<ProxyMutateRequest>> {
         let request = ProxyMutateRequest::Propose {
             proposal: Proposal {
-                id: Repr::new(id.clone()),
+                id: id.clone(),
```

Comment:
I think the old model is better, use of `Repr` should be reduced to just serialization/deserialization points.

```suggestion
        id: &ProposalId,
        author: &SigningKey,
        actions: &Vec<ProposalAction>,
    ) -> eyre::Result<Signed<ProxyMutateRequest>> {
        let request = ProxyMutateRequest::Propose {
            proposal: Proposal {
                id: id.rt().expect("infallible conversion"),
```

or

```suggestion
        id: &ProposalId,
        author: &SigningKey,
        actions: &Vec<ProposalAction>,
    ) -> eyre::Result<Signed<ProxyMutateRequest>> {
        let request = ProxyMutateRequest::Propose {
            proposal: Proposal {
                id: Repr::new(*id),
```